crowbar_framework/app/models/service_object.rb

```diff
@@ -1014,19 +1034,21 @@ def apply_role(role, inst, in_queue, bootstrap = false)
       new_deployment.delete("elements_expanded")
     end
 
-    # Build a list of old elements.
-    # elements_expanded on the old role is guaranteed to exists, as we already
-    # ran through apply_role with the old_role.  Cache is used for the case
-    # when pacemaker barclamp is deactivated.  elements_expanded gets updated
-    # by pacemaker barclamp.
-    old_elements = {}
-    old_deployment = old_role.override_attributes[@bc_name] unless old_role.nil?
-    unless old_deployment.nil?
-      old_elements = old_deployment["elements_expanded"]
-      if old_elements.nil?
-        old_elements = old_deployment["elements"]
+    # make sure the role is saved
+    role.save
```

Comment:
These two lines are wrong; I think that's due to a broken rebase.

Reply:
Something weird happened here as it showed like you never merged the patch that affected this, even with a rebase. Im guessing that it was a github delay or something like that. Fixed it!